docs/src/performance.md

```diff
@@ -0,0 +1,81 @@
+# Performance Profiling Tutorial
+
+## Effective Benchmarking with Hyperfine
+
+[Hyperfine](https://github.com/sharkdp/hyperfine) is a powerful command-line benchmarking tool that allows you to measure and compare execution times of commands with statistical rigor.
+
+### Benchmarking Best Practices
+
+When evaluating performance improvements, always set up your benchmarks to compare:
+
+1. The GNU implementation as reference
+2. The implementation without the change
+3. The implementation with your change
+
+This three-way comparison provides clear insights into:
+- How your implementation compares to the standard (GNU)
+- The actual performance impact of your specific change
+
+### Example Benchmark
+
+```bash
+# Three-way comparison benchmark
+hyperfine \
+  --warmup 3 \
+  "/usr/bin/ls -R ." \
+  "./target/release/coreutils.prev ls -R ." \
+  "./target/release/coreutils ls -R ."
```

Comment:
Found `-L` extremely useful when testing `seq`, so that you don't need to copy-paste the parameters:

```
hyperfine --warmup 3 -L ls /usr/bin/ls,"./target/release/coreutils.prev ls","./target/release/coreutil ls" "{ls} -R ."
```